services/report/languages/mono.py

```diff
@@ -22,23 +19,18 @@ def process(
 
 
 def from_xml(xml: Element, report_builder_session: ReportBuilderSession) -> Report:
-    path_fixer, ignored_lines = (
-        report_builder_session.path_fixer,
-        report_builder_session.ignored_lines,
-    )
+    files: dict[str, ReportFile | None] = {}
+
     # loop through methods
     for method in xml.iter("method"):
-        # get file name
-        filename = path_fixer(method.attrib["filename"])
-        if filename is None:
+        filename = method.attrib["filename"]
+        if filename not in files:
+            _file = report_builder_session.create_coverage_file(filename)
+            files[filename] = _file
+        _file = files[filename]
+        if _file is None:
```

Comment:
yes I think thats fine, I even think it might be good for perf as the path fixing can be slow sometimes. Some other language processors seem to cache that stuff, but usage is definitely not consistent.